### DISCUSSION TITLE
Limit date range for reporting only to realistic timeline

### DISCUSSION THREAD

**Discussion by juriansluiman** [firefly-iii/firefly-iii#5922] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/firefly-iii/about-firefly-iii/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been discussed before.

### Description

For reporting, the date range starts at 1970:

![afbeelding](https://user-images.githubusercontent.com/705925/159674681-72a0ecb5-1423-4626-8412-df4e9932cdf7.png)

All my transactions only exist after 2015, so I have to scroll along a very long list of non-realistic dates for my reports.

### Solution

Filter the date range by the earliest transaction. My transactions start at 01-01-2015, so having 2015 as first option would be reasonable.

### Additional context

I guess this filter won't cause any big performance problem as it's a single query to get the first result by transaction date. An improvement might be to index the transaction's dates if there's no index yet.

**Comment by JC5** [comment, marked as answer]
It already does that. You probably have a transaction with no date or a bad date somewhere.

**Reply by juriansluiman** [reply]
Oh nice, I just struggled with this for years, pushing myself to report this eventually :smiling_face_with_tear: Corrected a single transaction and things look much better now :+1:

**Reply by JC5** [reply]
Glad to hear it’s fixed! Don’t wait too long next time, I’m happy to help! 😅

**Reply by hickupper** [reply]
Had to export all my data to find it, but I found a start date on a liability dating back to year 0002. This helped immensely.
